wikidata_game/sanitizer.php

```diff
@@ -0,0 +1,95 @@
+<?php
+/**
+ * Dumps a db backup, and removes potential references from blacklisted sources
+ * This is throwaway code, meant to only run once as a resolution of bug T252781
+ */
+
+function getDb($dbhost, $dbname, $dbmycnf) {
+    $dbuser = $dbmycnf['user'];
+    $dbpass = $dbmycnf['password'];
+    return new PDO('mysql:host=' . $dbhost . ';dbname=' . $dbname . ';charset=utf8', $dbuser, $dbpass);
+}
+
+function backupTable($dbhost, $dbname, $mycnf_path) {
+    $root = dirname(__FILE__);
+    $backup_file = 'db-backup-ref-' . date('c') . '.sql';
+    exec("mysqldump --defaults-file=". $mycnf_path ." --host=$dbhost $dbname --result-file=$backup_file");
+}
+
+function getBlacklistedRefs($blacklist){
+    return function($row) use ($blacklist){
+        $data = json_decode($row["ref_data"], true);
```

Comment:
Double JSON decoding? c.f. line 55?

Reply:
Line 55 decodes the blacklist, this line decodes the reference data from the sql result

Reply:
doh! yep